### PR TITLE
Gjoranv/hit metrics histogram

### PR DESCRIPTION
**Pull request by gjoranv**
Please review carefully. Is it necessary to declare the Values explicitly, and is logging of the mean for totalhits desirable?

I have verified in a docker container, that the metrics will be output from the metrics-proxy.

### CODE REVIEW

**Review by baldersheim — APPROVED** [review]
This looks fine to me, but this is unfamiliar territory so I think that @arnej27959 might want to have a look too.

File: container-search/src/main/java/com/yahoo/prelude/statistics/StatisticsSearcher.java [hunk 1]
```diff
@@ -77,6 +77,9 @@
     private final Value peakQPS; // peak 1s QPS
     private final Counter emptyResults; // number of results containing no concrete hits
     private final Value hitsPerQuery; // mean number of hits per query
+    private final Value hitsPerQueryBuckets;
```
Comment (arnej27959) [inline container-search/src/main/java/com/yahoo/prelude/statistics/StatisticsSearcher.java:80]: the two new "*Buckets" values doesn't do anything useful here, you can just remove them.

**Review by arnej27959 — APPROVED** [review]
try removing the extra Values, otherwise looks good.

File: container-search/src/main/java/com/yahoo/prelude/statistics/StatisticsSearcher.java (hunk 1 above)
Comment (gjoranv) [reply container-search/src/main/java/com/yahoo/prelude/statistics/StatisticsSearcher.java]: Done. Doesn't this mean that also the `queryLatencyBuckets` value is redundant?

Comment (arnej27959) [reply container-search/src/main/java/com/yahoo/prelude/statistics/StatisticsSearcher.java]: queryLatency logs to "mean_query_latency" while queryLatencyBuckets is the one that logs to "query_latency".

actually, if you just want the new metrics in /state/v1/metrics (and so on) you should not need new Values at all; but they have the benefit of outputting their metric to vespa.log periodically which may be useful.

Comment (gjoranv) [reply container-search/src/main/java/com/yahoo/prelude/statistics/StatisticsSearcher.java]: Thanks for clarifying! There were several points I had missed here.

**Pull request merged**